src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/PersonalizeRankingResponseProcessor.java

```diff
@@ -0,0 +1,160 @@
+        PersonalizedRanker ranker = rankerFactory.getPersonalizedRanker(rankerConfig, personalizeClient);
+        long startTime = System.nanoTime();
+        SearchHits personalizedHits = ranker.rerank(hits);
+        long personalizeTimeTookMs = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime);
+
+        final SearchResponseSections transformedSearchResponseSections = new InternalSearchResponse(personalizedHits,
+                (InternalAggregations) response.getAggregations(), response.getSuggest(),
+                new SearchProfileShardResults(response.getProfileResults()), response.isTimedOut(),
+                response.isTerminatedEarly(), response.getNumReducePhases());
+
+        final SearchResponse transformedResponse = new SearchResponse(transformedSearchResponseSections, response.getScrollId(),
+                response.getTotalShards(), response.getSuccessfulShards(),
+                response.getSkippedShards(), response.getTook().getMillis() + personalizeTimeTookMs, response.getShardFailures(),
+                response.getClusters());
+
+        logger.info("Personalize ranking processor took " + personalizeTimeTookMs + " ms");
+
+        return transformedResponse;
+    }
+
+    /**
+     * Get the type of the processor.
+     */
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    /**
+     * Get the tag of a processor.
+     */
+    @Override
+    public String getTag() {
+        return tag;
+    }
+
+    /**
+     * Gets the description of a processor.
+     */
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    public static final class Factory implements Processor.Factory {
+
+        PersonalizeClientSettings personalizeClientSettings;
+
+        public Factory(PersonalizeClientSettings settings) {
+            this.personalizeClientSettings = settings;
+        }
+
+        @Override
+        public PersonalizeRankingResponseProcessor create(Map<String, Processor.Factory> processorFactories, String tag, String description, Map<String, Object> config) throws Exception {
+            // TODO: Handle validation as well as required vs non required config differentiation with related user error / exception
+            String personalizeCampaign = ConfigurationUtils.readStringProperty(TYPE, tag, config, CAMPAIGN_ARN_CONFIG_NAME);
+            String iamRoleArn = ConfigurationUtils.readStringProperty(TYPE, tag, config, IAM_ROLE_ARN_CONFIG_NAME);
+            String recipe = ConfigurationUtils.readStringProperty(TYPE, tag, config, RECIPE_CONFIG_NAME);
+            String itemIdField = ConfigurationUtils.readStringProperty(TYPE, tag, config, ITEM_ID_FIELD_CONFIG_NAME);
+            String awsRegion = ConfigurationUtils.readStringProperty(TYPE, tag, config, REGION_CONFIG_NAME);
+            double weight = Double.parseDouble(ConfigurationUtils.readStringProperty(TYPE, tag, config, WEIGHT_CONFIG_NAME));
```

Comment:
There's a `ConfigurationUtils.readDoubleProperty` that you could use here.

Reply:
Updated.